drivers/include/ltc4150.h

```diff
@@ -0,0 +1,347 @@
+ * Whenever this documentation refers to a breakout board, this open hardware
+ * board is meant. Of course, this driver works with the "bare" LTC4150 as well.
+ *
+ * Please note that this driver works interrupt driven and does not clear the
+ * signal. Thus, the /CLR and /INT pins on the LTC4150 need to be connected
+ * (in case of the breakout board: close solder jumper SJ1), so that the signal
+ * is automatically cleared.
+ *
+ * Hint: The breakout board uses external pull up resistors on /INT, POL and
+ * /SHDN. Therefore /SHDN can be left unconnected and no internal pull ups are
+ * required for /INT and POL. In case your board uses 3.3V logic the solder
+ * jumpers SJ2 and SJ3 have to be closed, in case of 5V they have to remain
+ * open. Connect the VIO pin to the logic level, GND to ground, IN+ and IN- to
+ * the power supply and use OUT+ and OUT- to power your board.
+ *
+ * In the easiest case only the /INT pin needs to be connected to a GPIO,
+ * and (in case of external pull ups) /SHDN and POL can be left unconnected.
+ * The GPIO /INT is connected to support for interrupts, /SHDN and POL
+ * (if connected) do not require interrupt support.
+ *
+ * In case a battery is used the POL pin connected to another GPIO. This allows
+ * to distinguish between charge drawn from the battery and charge transferred
+ * into the battery (used to load it).
+ *
+ * In case support to power off the LTC4150 is desired, the /SHDN pin needs to
+ * be connected to a third GPIO.
+ *
+ * # Things to keep in mind
+ * The LTC4150 creates pulses with a frequency depending on the current drawn.
+ * Thus, more interrupts need to be handled when more current is drawn, which
+ * in turn increases system load (and power consumption). The interrupt service
+ * routing is quite short and even when used outside of specification less than
+ * 20 ticks per second will occur. Hence, this effect should hopefully be
+ * negligible.
+ * @{
+ *
+ * @file
+ * @brief       LTC4150 coulomb counter
+ *
+ * @author      Marian Buschsieweke <marian.buschsieweke@ovgu.de>
+ */
+
+#ifndef LTC4150_H
+#define LTC4150_H
+
+#include <stdint.h>
+
+#include "mutex.h"
+#include "periph/gpio.h"
+#include "xtimer.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @brief Configuration flags of the LTC4150 coulomb counter
+ */
+enum {
+    /**
+     * @brief External pull on the /INT pin is present
+     */
+    LTC4150_INT_EXT_PULL_UP = 0x01,
+    /**
+     * @brief External pull on the /POL pin is present
+     */
+    LTC4150_POL_EXT_PULL_UP = 0x02,
+    /**
+     * @brief External pull on the /INT *and* the /POL pin is present
+     */
+    LTC4150_EXT_PULL_UP = LTC4150_INT_EXT_PULL_UP | LTC4150_POL_EXT_PULL_UP,
+};
+
+/**
+ * @brief Enumeration of directions in which the charge can be transferred
+ */
+typedef enum {
+    LTC4150_CHARGE,             /**< The battery is charged */
+    LTC4150_DISCHARGE,          /**< Charge is drawn from the battery */
+} ltc4150_dir_t;
```

Comment:
so its either `charge` or `discharge`, never both, right? So why not use a simple `bool charge` with `true` and `false` in the implementation. To me that would be more readable, e.g. by omitting `switch () case`.

Reply:
plus it saves 16 Bytes, when using `bool` instead of the `enum` ...

but I leave it to you, that's a non-blocking suggestion

Reply:
@MichelRottleuthner what do you think?

Reply:
Sidenote from afar: I like descriptive value names better.

Reply:
I also prefer the named values

Reply:
> plus it saves 16 Bytes, when using `bool` instead of the `enum` ...

This is a good point. I personally consider enums more readable than `bool`, but if someone wants to use that driver and her/his ROM is overflown by only a couple of bytes, this user would clearly favor the `bool` over the `enum`.

Honestly, I am totally surprised that there is a difference in code size by using a two-value `enum` over a `bool`. I'll have to look into this to understand that.

@miri64, @MichelRottleuthner, @smlng:

Would 

``` C
typedef bool ltc4150_dir_t;

#define LTC4150_CHARGE      true
#define LTC4150_DISCHARGE   false
```

be a compromise? (Sure, with the `enum` gdb would be able to show `LTC4150_CHARGE` instead of `1`...)

Reply:
I think the size difference comes from avoiding `switch ... case` when using bool

Reply:
but maybe, leave as if for the first integration of that driver and refine later? At least I wouldn't block if you stick with the enum for now.

Reply:
> I think the size difference comes from avoiding `switch ... case` when using bool

Yes, obviously. I should have seen that. I now optimized it by making the switch binary (by making one of the two `case`s the `default`), which reduced the ROM size by `28` bytes on my machine. With that, `enum` and `bool` should now indeed be the same regarding ROM size.